apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js

```diff
@@ -226,6 +226,47 @@ export const getStudentMCResponsesForCurrentAssessment = (state) => {
   return studentResponsesArray;
 };
 
+/**
+ * Returns an array of objects, each representing a free response question
+ * and all the students' responses to that question.
+ */
+export const getAssessmentsFreeResponseResults = (state) => {
+  const assessmentsStructure = getCurrentAssessmentQuestions(state);
+  if (!assessmentsStructure) {
+    return [];
+  }
+
+  // Initialize an object for each question, with the questionText and
+  // an empty array of responses.
+  const questionData = assessmentsStructure.questions;
+  const questionsAndResults = questionData.filter(question => question.type === 'FreeResponse').map(question => {
+    return {
+      questionText: question.question_text,
+      responses: [],
+    };
+  });
```

Comment:
Style nits: This starts with a fairly long line, and the map callback could have an expression function body:

```js
const questionsAndResults = questionData
  .filter(question => question.type === 'FreeResponse')
  .map(question => ({
    questionText: question.question_text,
    responses: [],
  });
```